WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+        Pattern emailRegExPattern = Patterns.EMAIL_ADDRESS;
+        Matcher matcher = emailRegExPattern.matcher(email);
+
+        return matcher.find() && email.length() <= MAX_EMAIL_LENGTH;
+    }
+
+    private final OnClickListener mNextClickListener = new OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            next(getEmail());
+        }
+    };
+
+    @Override
+    public void afterTextChanged(Editable s) {
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+        UpdateNextButton();
+        mEmailEditTextLayout.setError(null);
+    }
+
+    private void UpdateNextButton() {
+        if (fieldsFilled()) {
+            mNextButton.setEnabled(true);
+        } else {
+            mNextButton.setEnabled(false);
+        }
+    }
+
+    private boolean fieldsFilled() {
+        return EditTextUtils.getText(mEmailEditText).trim().length() > 0;
+    }
+
+    protected boolean isUserDataValid() {
+        final String email = EditTextUtils.getText(mEmailEditText).trim();
+        boolean retValue = true;
+
+        if (TextUtils.isEmpty(email)) {
+            showEmailError(R.string.required_field);
+            retValue = false;
+        }
+
+        return retValue;
+    }
+
+    private void showEmailError(int messageId) {
+        mEmailEditTextLayout.setError(getString(messageId));
+        mEmailEditText.post(new Runnable() {
+            @Override
+            public void run() {
+                mEmailEditText.requestFocus();
+            }
+        });
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        mDispatcher.register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        mDispatcher.unregister(this);
+    }
+
+    private void showEmailCheckProgressDialog() {
+        startProgress(getActivity().getString(R.string.checking_email));
+    }
+
+    @Override
+    protected void startProgress(String message) {
+        mNextButton.setEnabled(false);
+        mProgressDialog = ProgressDialog.show(getActivity(), "", message, true, true,
+                new DialogInterface.OnCancelListener() {
+            @Override
+            public void onCancel(DialogInterface dialog) {
+                UpdateNextButton();
+            }
+        });
+    }
+
+    @Override
+    protected void endProgress() {
+        if (mProgressDialog != null) {
+            mProgressDialog.cancel();
+        }
+
+        // nullify the reference to denote there is no operation in progress
+        mProgressDialog = null;
+
+        UpdateNextButton();
+    }
+
+    // OnChanged events
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAvailabilityChecked(OnAvailabilityChecked event) {
+        endProgress();
```

Comment:
I've isolated the email checking functionality into a headless retained fragment (`LoginEmailAvailabilityFragment`) with b352e5f.

Some implementation information: the logic supports multiple requests and only the latest one is respected and a result is returned. It also keeps the state so the client know if a request is pending or not. Rotations work well and the result cannot get lost while rotating.